src/server/pegasus_server_impl.cpp

```diff
@@ -2064,6 +2070,16 @@ int pegasus_server_impl::append_key_value_for_scan(
         return 2;
     }
 
+    if (request_validate_hash && _validate_partition_hash) {
+        if (_partition_version < 0 || _gpid.get_partition_index() > _partition_version ||
+            !check_pegasus_key_hash(key, _gpid.get_partition_index(), _partition_version)) {
+            if (_verbose_log) {
+                derror_replica("not serve hash key while scan");
+            }
+            return 4;
```

Comment:
What's the meaning of 1 2 3 and 4? could you make it a enum class?

Reply:
This is a history problem, you can reference `.h` file, I will refactor in next pull request, this pr only contains functional code.